### DISCUSSION TITLE
NEXT_PUBLIC_KEY is populated with both - development - & production secret

### DISCUSSION THREAD

**Discussion by unitokala** [vercel/next.js#20591] [Help]
I have two environment files for each stage inside the _root directory_ of my project as I need to connect to my backend on localhost during development. In these files I am exposing the **same key** - NEXT_PUBLIC_MYKEY, but with **different values for each environment.** I dont use any other env files in my directory.
- **.env.development**    
-- `NEXT_PUBLIC_MYKEY=http://localhost:8080`
- **.env.production**          
--` NEXT_PUBLIC_MYKEY=8.8.8.8`

While developing local i dont have no problems but by recently checking the **deployed version** on vercel I see the connection string to my backend is **merged together from development & production**. The client is now trying to connect to `http://localhost:8080/8.8.8.8/login`.

Did somebody else encountered this problem?

**Comment by lfades** [comment]
Please provide a reproduction that we can take a look at. We have tests to ensure the behavior you're describing doesn't happen 🙏
